Source/Layer.js

```diff
@@ -7,317 +7,405 @@
+  */
 
-    /**
-      Add a new wrapped layer object to represent a Sketch layer.
-      Apply any supplied properties to the wrapper (which will apply
-      them in turn to the wrapped layer).
+  get isImage() { return false; }
 
-      @param {MSLayer} newLayer The underlying Sketch layer object.
-      @param {dictionary} properties The properties to apply.
-      @param {string} wrapper The name of wrapper class to use.
-      @return {Layer} The wrapped layer object.
-    */
+  /**
+  Add a new wrapped layer object to represent a Sketch layer.
+  Apply any supplied properties to the wrapper (which will apply
+  them in turn to the wrapped layer).
 
-    _addWrappedLayerWithProperties(newLayer, properties, wrapper) {
-      if (newLayer) {
-        // add the Sketch object to this layer
-        var layer = this._object
-        layer.addLayers_(NSArray.arrayWithObject_(newLayer))
+  @param {MSLayer} newLayer The underlying Sketch layer object.
+  @param {dictionary} properties The properties to apply.
+  @param {string} wrapper The name of wrapper class to use.
+  @return {Layer} The wrapped layer object.
+  */
 
-        // make a Javascript wrapper object for the new layer
-        var wrapper = this._document.wrapObject(newLayer)
+  _addWrappedLayerWithProperties(newLayer, properties, wrapper) {
+    if (newLayer) {
+      // add the Sketch object to this layer
+      var layer = this._object
+      layer.addLayers_(NSArray.arrayWithObject_(newLayer))
 
-        // apply properties, via the wrapper
-        for (var p in properties) {
-          wrapper[p] = properties[p]
-        }
+      // make a Javascript wrapper object for the new layer
+      var wrapper = this._document.wrapObject(newLayer)
 
-        return wrapper
+      // apply properties, via the wrapper
+      for (var p in properties) {
+        wrapper[p] = properties[p]
       }
+
+      return wrapper
+    }
+  }
+
+  /**
+  Extract the frame to use for a layer from some properties.
+  If the frame wasn't supplied in the properties, we return a default value instead.
+
+  @param {dictionary} properties The properties to use when looking for a frame value.
+  @return {Rectangle} The frame rectangle to use.
+  */
+
+  _frameForLayerWithProperties(properties) {
+    var frame = properties.frame
+    if (frame) {
+      delete properties["frame"]
+    } else {
+      frame = new Rectangle(0, 0, 100, 100)
+    }
+    return frame
+  }
+
+  /**
+  Extract the style to use for a layer from some properties.
+  If the style wasn't supplied at all, we use the default one.
+  */
+
+  _styleForLayerWithProperties(properties) {
+    var style = properties.style
+    if (!style) {
+      style = new Style()
     }
 
-    /**
-      Extract the frame to use for a layer from some properties.
-      If the frame wasn't supplied in the properties, we return a default value instead.
+    var fills = properties.fills
+    if (fills) {
+      delete properties["fills"]
```

Comment:
I'm confused by this `properties.fills` vs `properties["fills"]`, but something tells me this is one of those stupid things JavaScript does, so feel free to ignore the comment